src/s2geography/geoarrow_test.cc

```diff
@@ -202,3 +205,49 @@ TEST(GeoArrow, GeoArrowReaderReadWKTCollection) {
   EXPECT_EQ(linestring->edge(0).v0, S2LatLng::FromDegrees(1, 0).ToPoint());
   EXPECT_EQ(linestring->edge(0).v1, S2LatLng::FromDegrees(3, 2).ToPoint());
 }
+
+using s2geography::geoarrow::Writer;
+
+TEST(GeoArrow, GeoArrowWriterPoint) {
+  s2geography::WKTReader reader;
+  auto geog1 = reader.read_feature("POINT (0 1)");
+  auto geog2 = reader.read_feature("POINT (2 3)");
+
+  Writer writer;
+  nanoarrow::UniqueSchema schema;
+  nanoarrow::UniqueArray array;
+
+  InitSchemaGeoArrowPoint(schema.get());
+  writer.Init(schema.get());
+
+  writer.WriteGeography(*geog1);
+  writer.WriteGeography(*geog2);
+  writer.Finish(array.get());
+
+  EXPECT_EQ(array->length, 2);
+  EXPECT_THAT(nanoarrow::ViewArrayAs<double_t>(array->children[0]), ElementsAre(0.0, 2.0));
```

Comment:
@paleolimbot any idea what could be wrong here from a quick glance? 

This is failing, but I have to say that the error message from gmock / ElementsAre is not great (printing actual / expected bytes instead of numbers)

Reply:
> I think you can do `ElementsAre(DoubleEq(0.0), DoubleEq(2.0))`

I actually had already tried testing that, but got some compilation errors:

```
/home/joris/scipy/repos/s2geography/src/s2geography/geoarrow_test.cc:228:3:   required from here
/home/joris/scipy/repos/s2geography/build/_deps/googletest-src/googletest/include/gtest/gtest-matchers.h:184:14: error: no match for 'operator==' (operand types are 'const nanoarrow::internal::Maybe<double>' and 'const testing::internal::FloatingEqMatcher<double>')
  184 |     return a == b;
      |            ~~^~~~
```

Looking at it more closely now, it seems that the usage of nanoarrow's `Maybe` (through `ViewArrayAs`) doesn't play nicely with `DoubleEq` from gmock